### PR TITLE
Snap notifications integration

### LINKED ISSUE #14604: Integrate new `NotificationController` into the extension
Hook up the new `NotificationController` to the `snap_notify` RPC method and implement the UI to view these notifications.

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## Explanation

- [x] Hook NotificationController to metamask and persist the controller's state
- [x] Create selectors and actions to manage notifications
- [x] Add notifications UI

## More Information

* Fixes #14604

## Manual Testing Steps

- Spin up the [Notification example snap](https://github.com/MetaMask/snaps-skunkworks/tree/main/packages/examples/examples/notifications)
- Send a notification to MetaMask
- Read the notification
- Clear the notifications

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] Manual testing complete & passed
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** QA attention is required, "QA Board" label has been applied
- [ ] PR has been added to the appropriate release Milestone

### CODE REVIEW

**Comment by eriknson** [comment]
Just reviewed Snaps notifications with @GuillaumeRx and it's smooth!

**Comment by FrederikBolding** [comment]
This should now be unblocked as #14700 was just merged!

File: ui/components/app/account-menu/account-menu.component.js [hunk 1]
```diff
@@ -400,6 +409,33 @@ export default class AccountMenu extends Component {
...
+            <AccountMenuItem
+              onClick={() => {
+                toggleAccountMenu();
+                history.push(NOTIFICATIONS_ROUTE);
+              }}
+              icon={
+                <div className="account-menu__notifications">
+                  <i
+                    className="fa fa-bell fa-xl"
+                    color="var(--color-icon-default)"
```
Comment (darkwing) [inline ui/components/app/account-menu/account-menu.component.js:424]: Does this `color` property work on a native HTML element?

Comment (GuillaumeRx) [reply ui/components/app/account-menu/account-menu.component.js:424]: Yes it works on the font-awesome icons 👍

File: ui/pages/notifications/notifications.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  );
+}
+
+NotificationItem.propTypes = {
+  notification: {
+    id: PropTypes.string.isRequired,
+    message: PropTypes.string.isRequired,
+    origin: PropTypes.string.isRequired,
+    createdDate: PropTypes.number.isRequired,
+    readDate: PropTypes.number.isRequired,
```
[the remark this answers is not in the document]
Comment (GuillaumeRx) [reply ui/pages/notifications/notifications.js:138]: It's set to `null` as default, I don't think that makes it optional ?

File: ui/pages/notifications/notification.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          sectionMessage: () => 'test',
+          route: '/test',
+          icon: 'test',
+        },
+      ],
+      onItemClick: jest.fn(),
+    };
+    const { getByText } = render(props);
+
+    expect(getByText(props.notification.message)).toBeDefined();
```
[the remark this answers is not in the document]
Comment (GuillaumeRx) [reply ui/pages/notifications/notification.test.js:98]: I tried but I can't manage to get this sentence. I feel like it has something to do with translations

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
This is excellent!  Tried it out and was awesome!  Well done!

**Pull request merged**